From 2db880244c42007634f81101e8a795ddd8ee62ca Mon Sep 17 00:00:00 2001
From: Tom <cryptite@gmail.com>
Date: Fri, 12 Jul 2019 07:59:35 -0500
Subject: [PATCH] Don't send CommandMap to players on world change

Often, but not 100% of the time, the server will reconstruct and recalculate the Command Map sent to players
when they change worlds. This is often laggy; worse-so depending on the extent of the tab-completed commands to be sent.
In our case, the map sent to the player on login is sufficient and is not tossed by the client for world changes.

As a result, we opt not to send this on world change and this prevents a fairly heavy, and consistent main-thread
lagspike whenever players change worlds, which can be a lot.

diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index 012238ac..c000d713 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -729,7 +729,7 @@ public abstract class PlayerList {
         entityplayer1.playerConnection.sendPacket(new PacketPlayOutSpawnPosition(blockposition1));
         entityplayer1.playerConnection.sendPacket(new PacketPlayOutExperience(entityplayer1.exp, entityplayer1.expTotal, entityplayer1.expLevel));
         this.b(entityplayer1, worldserver);
-        this.f(entityplayer1);
+//        this.f(entityplayer1); //Do not send/rebuild CommandMaps to players on world change. In our case, they don't change per-world. On-login is sufficient
         if (!entityplayer.playerConnection.isDisconnected()) {
             worldserver.getPlayerChunkMap().addPlayer(entityplayer1);
             worldserver.addEntity(entityplayer1);
-- 
2.17.1.windows.2

